src/canvas/BlendMode.js

```diff
@@ -212,6 +212,90 @@ var BlendMode = new function() {
             dr = 255 - abs(255 - sr - br);
             dg = 255 - abs(255 - sg - bg);
             db = 255 - abs(255 - sb - bb);
+        },
+
+        divide: function() {
+            dr = min (255, 255 * sr / (br + 1));
+            dg = min (255, 255 * sg / (bg + 1));
+            db = min (255, 255 * sb / (bb + 1));
+        },
+
+        'divide-source': function() {
+            dr = min (255, 255 * br / (sr + 1));
+            dg = min (255, 255 * bg / (sg + 1));
+            db = min (255, 255 * bb / (sb + 1));
+        },
+
+        'hard-mix': function() {
+            dr = sr + br < 255 ? 0 : 255;
+            dg = sg + bg < 255 ? 0 : 255;
+            db = sb + bb < 255 ? 0 : 255;
+        },
+
+        'pegtop-light': function() {
+            dr = (2 * sr * br + sr ** 2 * (1 - br / 127.5)) / 255;
```

Comment:
I am afraid `sr ** 2` is es2016 feature, thus, `sr * sr` might be better. http://kangax.github.io/compat-table/es2016plus/

Reply:
Totally agreed, that's now going to work for many browsers. `sr * sr` will though!